crates/mpz-ole-core/src/receiver.rs

```diff
@@ -0,0 +1,143 @@
+//! Receiver implementation.
+
+use crate::{
+    core::{ReceiverAdjust, ReceiverShare, ShareAdjust},
+    msg::{BatchAdjust, MaskedInputs},
+    OLEError, TransferId,
+};
+use mpz_fields::Field;
+use std::collections::VecDeque;
+
+/// A receiver for batched OLE.
+#[derive(Debug)]
+pub struct OLEReceiver<const N: usize, F> {
+    id: TransferId,
+    cache: VecDeque<ReceiverShare<F>>,
+}
+
+impl<const N: usize, F: Field> Default for OLEReceiver<N, F> {
+    fn default() -> Self {
+        OLEReceiver {
+            id: TransferId::default(),
+            cache: VecDeque::default(),
+        }
+    }
+}
+
+impl<const N: usize, F: Field> OLEReceiver<N, F> {
+    /// Generates new OLEs and stores them internally.
+    ///
+    /// # Arguments
+    ///
+    /// * `input` - The receiver's OLE input factors.
```

Comment:
Let's call them shares like in the core module

```suggestion
    /// * `input` - The receiver's OLE input shares.
```